### ISSUE TITLE
Chart is not rendered in dashboard but is when viewed alone

### ISSUE THREAD

**Issue by EuanKirkhope** [metabase/metabase#34998]
### Describe the bug

Viewing a dashboard shows an empty chart with incorrect catagories on the axis / legend.
Clicking into the chart to view alone it renders correctly.

Changing the chart to another chart type from stacked bar works, but it can never be put back as a stacked chart.

### To Reproduce

See the video
[screen-capture.webm](https://github.com/metabase/metabase/assets/12889375/edf21345-84ba-4cd6-bfc4-aef7785a2fc0)

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "17.0.7+7-LTS",
    "java.vendor": "Microsoft",
    "java.vendor.url": "https://www.microsoft.com",
    "java.version": "17.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "17.0.7+7-LTS",
    "os.name": "Windows 11",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "Australia/Sydney"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-10-20",
      "tag": "v0.47.5",
      "branch": "?",
      "hash": "e7ee12e"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

This is a show stopper, along with the dodgy email reports.

**Comment by paoliniluis** [comment]
Thank you! do you see any error in the browser console when you're on the dashboard?

**Comment by EuanKirkhope** [comment]
> Thank you! do you see any error in the browser console when you're on the dashboard?

![image](https://github.com/metabase/metabase/assets/12889375/992cdc31-b2d0-40c1-a701-6a30785fe41e)

**Comment by notz** [comment]
We experience the same problem sice we upgraded from 47.4 to 47.5:

```
TypeError: Cannot read properties of undefined (reading 'source')
    at SVGGElement.<anonymous> (timelines.js:155:26)
    at SVGGElement.<anonymous> (d3.js:792:21)
    at d3.js:962:16
    at Ae (d3.js:968:30)
    at K.each (d3.js:961:12)
    at K.html (d3.js:791:36)
    at timelines.js:153:6
    at hh (timelines.js:240:3)
    at LineAreaBarPostRender.js:402:3
    at xb (LineAreaBarPostRender.js:453:3)
value @ CardRenderer.jsx:97
```

If we disable the "date" x-display the graphs are working:

![image](https://github.com/metabase/metabase/assets/697251/0ee2a2ff-6f3a-46fd-b007-b7e7a9577877)

Perhaps it has something todo with events. Because if i move the graph to another collection (without events) they are working again.

**Comment by sicarul** [comment]
I'm seeing the same behavior as @notz on some questions:
<img width="1359" alt="image" src="https://github.com/metabase/metabase/assets/4952116/eb033106-d722-4228-a7f5-12c6ea1c0a2a">

If i change to ordinal or hide the lines and marks, it works.
<img width="1869" alt="image" src="https://github.com/metabase/metabase/assets/4952116/7eca8384-8dbb-406c-85f0-7e67c4843b51">

**Comment by notz** [comment]
The bug seems to be a duplicate of #35129 and is fixed with version 0.47.7

**Comment by TVanneufvilleJoko** [comment]
I'm getting the same issue on v48. Any news from Metabase?

**Comment by darksciencebase** [comment]
@TVanneufvilleJoko can you provide the steps to reproduce? The original issue has been fixed and shipped in 47.7.
